### ISSUE TITLE
Application Autoscaling resources not deploying in AWS GovCloud

### ISSUE THREAD

**Issue by jmhale** [terraform-providers/terraform-provider-aws#7067]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

`Terraform v0.11.11`
`* provider.aws: version = "~> 1.54"`

### Affected Resource(s)

* `aws_appautoscaling_target`
* `aws_appautoscaling_policy`

### Terraform Configuration Files

```hcl
 resource "aws_appautoscaling_target" "dynamodb_table_write_target" {
   max_capacity       = 10000
   min_capacity       = 10
   resource_id        = "table/${aws_dynamodb_table.odd.name}"
   role_arn           = "${aws_iam_role.odd_lambda_exec_role.arn}"
   scalable_dimension = "dynamodb:table:WriteCapacityUnits"
   service_namespace  = "dynamodb"
 }

 resource "aws_appautoscaling_policy" "dynamodb_table_write_policy" {
   name               = "DynamoDBWriteCapacityUtilization:${aws_appautoscaling_target.dynamodb_table_read_target.resource_id}"
   policy_type        = "TargetTrackingScaling"
   resource_id        = "${aws_appautoscaling_target.dynamodb_table_write_target.resource_id}"
   scalable_dimension = "${aws_appautoscaling_target.dynamodb_table_write_target.scalable_dimension}"
   service_namespace  = "${aws_appautoscaling_target.dynamodb_table_write_target.service_namespace}"

   target_tracking_scaling_policy_configuration {
     predefined_metric_specification {
       predefined_metric_type = "DynamoDBWriteCapacityUtilization"
     }

     target_value = 60
   }
 }
```

### Expected Behavior

`aws_appautoscaling_target` & `aws_appautoscaling_policy` resources should be created.

### Actual Behavior

`aws_appautoscaling_target` & `aws_appautoscaling_policy` resources are not created with a "host not found" error.

### Steps to Reproduce

1. Obtain credentials in AWS GovCloud.
2. `terraform apply`

### Important Factoids

This appears to be related to #4690, except it's happening for GovCloud. 

The aws-sdk-go library is returning the wrong hostname for the autoscaling API in GovCloud. 

https://application-autoscaling.us-gov-west-1.amazonaws.com/, when we should be getting https://autoscaling.us-gov-west-1.amazonaws.com/

### References

Upstream issue: https://github.com/aws/aws-sdk-go/issues/2391

**Comment by bflad** [comment]
Let's ask upstream if they'll accept a PR similar to https://github.com/aws/aws-sdk-go/pull/2080 👍

**Comment by bflad** [comment]
The AWS Go SDK update for this has been merged and will release with version 1.56.0 of the Terraform AWS provider, likely in the next day or two. 🎉
